packages/babel-runtime-corejs3/package.json

```diff
@@ -353,6 +353,7 @@
     "./helpers/interopRequireDefault": [
       {
         "node": "./helpers/interopRequireDefault.js",
+        "react-native": "./helpers/interopRequireDefault.js",
         "import": "./helpers/esm/interopRequireDefault.js",
         "default": "./helpers/interopRequireDefault.js"
```

Comment:
I'm not sure I understand what's going on here. Why doesn't react native just load the `default` fallback?

Reply:
(PR description incoming! 🔥)

We're launching Package Exports support in React Native experimentally very soon, and the first implementation of it does not yet dynamically switch between `"import"`/`"require"` conditions during resolution (as our bundler, Metro, does not yet handle the ESM spec) — we match **both** `"import"` and `"require"` in an effort to maximise library compatibility for users.
- This works in practice due to the use of Babel's `interopRequireDefault` transform allowing module types to work interchangeably.
- (This is conceptually similar to TypeScript's [`--moduleResolution bundler`](https://github.com/microsoft/TypeScript/pull/51669).)

Reply:
The motivation is actually very similar to #12865 — I'm going to retitle and update this PR to include the `"require"` condition (keeping `"import"`, but ordering before this) as a preferred fix.

Reply:
(Resolving conversation for updated PR)